src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -1275,14 +1277,24 @@ impl EncodeContext<'tcx> {
         record!(self.per_def.visibility[def_id] <- ty::Visibility::Public);
         record!(self.per_def.span[def_id] <- macro_def.span);
         record!(self.per_def.attributes[def_id] <- macro_def.attrs);
+        self.encode_ident_span(def_id, macro_def.ident);
         self.encode_stability(def_id);
         self.encode_deprecation(def_id);
     }
 
-    fn encode_info_for_generic_param(&mut self, def_id: DefId, kind: EntryKind, encode_type: bool) {
+    fn encode_info_for_generic_param(
+        &mut self,
+        def_id: DefId,
+        kind: EntryKind,
+        name: ParamName,
+        encode_type: bool,
+    ) {
         record!(self.per_def.kind[def_id] <- kind);
         record!(self.per_def.visibility[def_id] <- ty::Visibility::Public);
         record!(self.per_def.span[def_id] <- self.tcx.def_span(def_id));
+        if let ParamName::Plain(ident) = name {
+            self.encode_ident_span(def_id, ident);
```

Comment:
Looks like this span is never decoded?
Could you audit the uses of `encode_ident_span` and avoid encoding spans that are never decoded, they only spend extra compilation time.